File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
import collections

def solution(cacheSize, cities):
    cache = collections.deque(maxlen=cacheSize)
    time = 0
    
    for city in cities:
        city = city.lower()
        if city in cache:
            cache.remove(city)
            cache.append(city)
            time += 1
        else:
            cache.append(city)
            time += 5
            
    return time

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.js
function solution(cacheSize, cities) {
    const MISS = 5, HIT = 1;

    if (!cacheSize) return MISS * cities.length;

    let time = 0,
        cache = [];

    cities.forEach(city => {
        city = city.toLowerCase();

        let idx = cache.indexOf(city);

        if (idx !== -1) {
            cache.splice(idx, 1);
            time += HIT;
        } else {
            if (cache.length === cacheSize) cache.shift();
            time += MISS;
        }
        cache.push(city);
    });

    return time;
}